generated/nidcpower/session.py

```diff
@@ -2855,6 +2870,60 @@ def _get_error(self):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=True)
         return int(code_ctype.value), description_ctype.value.decode(self._encoding)
 
+    def lock_session(self, caller_has_lock=None):
```

Comment:
I don't think this `caller_has_lock` business is useful to Python users. My understanding from the documentation is that it's useful to a C user that has no access to things like RAII or context managers and is using `checkErr` macros with their fun gotos.

My vote is to expose only `def lock_session(self)` that returns the context manager and doesn't have `caller_has_lock` and `def unlock_session(self)` with no parameters.

Another thing to consider is whether we want to make these functions match, for duck-typing reasons, the naming used in rlock. Need to think about that a little bit more.